azblob/zt_policy_request_id_test.go

```diff
@@ -0,0 +1,101 @@
+package azblob
+
+import (
+	"context"
+	"errors"
+	"github.com/Azure/azure-pipeline-go/pipeline"
+	chk "gopkg.in/check.v1"
+	"net/http"
+	"net/url"
+)
+
+type requestIDTestScenario int
+
+const (
+	// Testing scenarios for echoing Client Request ID
+	clientRequestIDMissing requestIDTestScenario = 1
+	clientRequestIDError   requestIDTestScenario = 2
+	clientRequestIDMatch   requestIDTestScenario = 3
+	clientRequestIDNoMatch requestIDTestScenario = 4
+)
+
+type dummyPolicy struct {
+	matchID  string
+	scenario requestIDTestScenario
+}
+
+func (p dummyPolicy) Do(ctx context.Context, request pipeline.Request) (pipeline.Response, error) {
+	var header http.Header = make(map[string][]string)
+	var err error
+
+	// Set headers and errors according to each scenario
+	switch p.scenario {
+	case clientRequestIDMissing:
+		// header.Add("x-ms-client-request-id", "")
+	case clientRequestIDError:
+		err = errors.New("error is not nil")
+	case clientRequestIDMatch:
+		header.Add("x-ms-client-request-id", p.matchID)
```

Comment:
For clientRequestIDMatch, couldn't you get the request ID from the argument `request`? Instead of using a variable on the struct.